### ISSUE TITLE
Testing update of Tendermint configuration

### ISSUE THREAD

**Issue by Karrenbelt** [valory-xyz/open-autonomy#1364]
Right now in e2e test for registration start up we exchange tendermint configurations, update the genesis file, and restart the tendermint nodes. However, since Tendermint is already running prior to this and the genesis file already exists containing the information of the other nodes, the tests offer no guarantee that the updating and restarting is working in accordance with our expectations. We thus should add a test to see that we can indeed restart the network with an altered genesis file. 

On a side note, we will possibly transition to a setup where Tendermint is no running before registration start-up in the near future; it certainly will not be expected to have the necessary information on the other nodes that is currently present before starting.

**Comment by DavidMinarsch** [comment]
Closing as addressed.

The e2e test (`packages/valory/agents/registration_start_up/tests/test_registration.py`) now exercises the full happy path and asserts both the update invocation and post-restart sync via strict log checks:

```python
STRICT_CHECK_STRINGS = (
    log_messages.request_personal.value,
    log_messages.response_personal.value,
    log_messages.request_verification.value,
    log_messages.response_verification.value,
    log_messages.request_others.value,
    log_messages.collection_complete.value,
    log_messages.request_update.value,
    log_messages.response_update.value,
    "local height == remote height == ",
    "; continuing execution...",
)
```

Runs with 4 nodes against real tendermint containers with `share_tm_config_on_startup=True`, so the test covers: genesis exchange → config update → tendermint restart with altered genesis → post-restart height alignment.

The 2022 side-note about transitioning to a setup where Tendermint doesn't run before registration is a separate design choice and unchanged today; that belongs in its own ticket if pursued.
